server/app.mjs

```diff
@@ -34,38 +37,54 @@ export const startAppServer = ({dev, hostname, port, dir}) => {
         });
         const handler = app.getRequestHandler();
 
-        app.prepare().then(() => {
-            const httpServer = createServer(handler);
-            const io = new Server(httpServer);
-            const gptscript = new GPTScript({disableCache: process.env.DISABLE_CACHE === "true"});
-
-            io.on("connection", (socket) => {
-                io.emit("message", "connected");
-                socket.on("run", async (location, tool, args, scriptWorkspace, threadID) => {
-                    if (runningScript) {
-                        await runningScript.close();
-                        runningScript = null;
-                    }
-                    try {
-                        dismount(socket);
-                        await mount(location, tool, args, scriptWorkspace, socket, threadID, gptscript);
-                    } catch (e) {
-                        socket.emit("error", e);
-                    }
+        const gptscript = new GPTScript({disableCache: process.env.DISABLE_CACHE === "true"});
+
+        if (!gptscriptInitialized) {
+            gptscriptInitPromise = initGPTScriptConfig(gptscript)
+                .then(() => {
+                    gptscriptInitialized = true;
+                    console.log('GPTScript config initialized');
+                })
+                .catch((err) => {
+                    console.error('Error initializing GPTScript config:', err);
+                    reject(err);
+                    return;
```

Comment:
Does anyone know if I need to create a new `GPTScript` object to get it to use the updated config after this step?

Reply:
In general, yes. But it doesn't seem like you are changing any of the config for the `GPTScript` object itself.